src/compiler/checker.ts

```diff
@@ -12191,9 +12192,7 @@ namespace ts {
                             }
                         }
                     }
-                    if (elementTypes.length) {
-                        return createTupleType(elementTypes);
-                    }
+                    return createTupleType(elementTypes);

```

Comment:
I think you should update the `createTupleType` function to return the `emptyTupleType` singleton if `elementTypes` is empty.